packaging/kibana_plugin.py

```diff
@@ -0,0 +1,231 @@
+    returned: success
+    type: string
+state:
+    description: the state for the managed plugin
+    returned: success
+    type: string
+'''
+
+PACKAGE_STATE_MAP = dict(
+    present="--install",
+    absent="--remove"
+)
+
+def parse_plugin_repo(string):
+    elements = string.split("/")
+
+    # We first consider the simplest form: pluginname
+    repo = elements[0]
+
+    # We consider the form: username/pluginname
+    if len(elements) > 1:
+        repo = elements[1]
+
+    # remove elasticsearch- prefix
+    # remove es- prefix
+    for string in ("elasticsearch-", "es-"):
+        if repo.startswith(string):
+            return repo[len(string):]
+
+    return repo
+
+def is_plugin_present(plugin_dir, working_dir):
+    return os.path.isdir(os.path.join(working_dir, plugin_dir))
+
+def parse_error(string):
+    reason = "reason: "
+    try:
+        return string[string.index(reason) + len(reason):].strip()
+    except ValueError:
+        return string
+
+def install_plugin(module, plugin_bin, plugin_name, url, timeout):
+    cmd_args = [plugin_bin, "plugin", PACKAGE_STATE_MAP["present"], plugin_name]
+
+    if url:
+        cmd_args.append("--url %s" % url)
+
+    if timeout:
+        cmd_args.append("--timeout %s" % timeout)
+
+    cmd = " ".join(cmd_args)
+
+    rc, out, err = module.run_command(cmd)
+
+    if rc != 0:
+        reason = parse_error(out)
+        module.fail_json(msg=reason)
+    
+    return True, cmd, out, err
+
+def remove_plugin(module, plugin_bin, plugin_name):
+    cmd_args = [plugin_bin, "plugin", PACKAGE_STATE_MAP["absent"], plugin_name]
+
+    cmd = " ".join(cmd_args)
+
+    rc, out, err = module.run_command(cmd)
+
+    if rc != 0:
+        reason = parse_error(out)
+        module.fail_json(msg=reason)
+    
+    return True, cmd, out, err
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            state=dict(default="present", choices=PACKAGE_STATE_MAP.keys()),
+            url=dict(default=None),
+            timeout=dict(default="1m"),
+            plugin_bin=dict(default="/opt/kibana/bin/kibana", type="path"),
+            plugin_dir=dict(default="/opt/kibana/installedPlugins/", type="path"),
+            version=dict(default=None),
+            force=dict(default="no", type="bool")
+        )
+    )
+
+    name        = module.params["name"]
+    state       = module.params["state"]
+    url         = module.params["url"]
+    timeout     = module.params["timeout"]
+    plugin_bin  = module.params["plugin_bin"]
+    plugin_dir  = module.params["plugin_dir"]
+    version     = module.params["version"]
+    force       = module.params["force"]
+
+    present = is_plugin_present(parse_plugin_repo(name), plugin_dir)
+
+    # skip if the state is correct
+    if (present and state == "present" and not force) or (state == "absent" and not present and not force):
+        module.exit_json(changed=False, name=name)
```

Comment:
Looks like adding support for check mode should be straightforward.

See http://docs.ansible.com/ansible/playbooks_checkmode.html#check-mode-dry-run